config/m4/compiler.m4

```diff
@@ -195,6 +195,25 @@ AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
                 CFLAGS="$SAVE_CFLAGS"])
 
 
+#
+# Set default datatype alignment to 16 bytes.
+# Some compilers (LLVM based, clang) expects allocation of datatypes by 32 bytes
+# to optimize operations memset/memcpy/etc using vectorized processor instructions
+# which requires aligment of memory buffer by 32 or higer bytes. Default malloc method
+# guarantee alignment for 16 bytes only. Force using compiler 16-bytes alignment
+# by default if option is supported.
+#
+SAVE_CFLAGS="$CFLAGS"
+CFLAGS="$BASE_CFLAGS $CFLAGS -fmax-type-align=16"
+AC_MSG_CHECKING([-fmax-type-align=16])
+AC_COMPILE_IFELSE([AC_LANG_SOURCE([[int main(){}]])],
+                  [AC_MSG_RESULT([yes])
+                   BASE_CFLAGS="$BASE_CFLAGS -fmax-type-align=16"
+                   AC_DEFINE([UCX_ALLOC_ALIGN], 16, [Set aligment assumption for compiler])],
```

Comment:
can we define it outside the AC_COMPILE_IFELSE([..]), and do something like:
`BASE_CFLAGS="$BASE_CFLAGS -fmax-type-align=$UCX_ALLOC_ALIGN"`
(not required, only if it can work with autoconf..)

Reply:
ok, done